src/modules/netinstall/NetInstallPage.cpp

```diff
@@ -0,0 +1,139 @@
+        YAML::Node groupDefinition = it->as<YAML::Node>();
+
+        QString name( tr( groupDefinition["name"].as<std::string>().c_str() ) );
+        QString description( tr( groupDefinition["description"].as<std::string>().c_str() ) );
+        QStringList packages;
+
+        for ( YAML::const_iterator it = groupDefinition["packages"].begin();
+                it != groupDefinition["packages"].end(); ++it )
+            packages.append( ( *it ).as<std::string>().c_str() );
+
+        m_groups[name].name = name;
+        m_groups[name].description = description;
+        m_groups[name].packages = packages;
+
+        if ( groupDefinition["selected"] )
+            m_groups[name].selected = groupDefinition["selected"].as<bool>();
+
+        if ( groupDefinition["hidden"] )
+            m_groups[name].hidden = groupDefinition["hidden"].as<bool>();
+
+        m_groupOrder.append( name );
+    }
+}
+
+void
+NetInstallPage::dataIsHere( KJob* job )
+{
+    if ( job->error() )
+    {
+        qDebug() << job->errorString();
+        ui->netinst_status->setText( tr( "Network Installation. (Disabled: Unable to fetch package lists, check your network connection)" ) );
+        return;
+    }
+
+    auto transferJob = dynamic_cast<KIO::StoredTransferJob*>( job );
+    Q_ASSERT( transferJob != nullptr );
+    ReadGroups( transferJob->data() );
+
+    QSignalMapper* mapper = new QSignalMapper( this );
+    foreach ( const QString& groupKey, m_groupOrder )
+    {
+        Group group = m_groups[groupKey];
+        if ( group.hidden )
+        {
+            // Do not present on view.
+            continue;
+        }
+
+        GroupSelectionWidget* groupWidget = new GroupSelectionWidget( group.name, group.description, group.packages, this );
+        m_groupWidgets.insert( groupKey, groupWidget );
+        ui->groupswidget->layout()->addWidget( groupWidget );
+
+        mapper->setMapping( groupWidget, groupKey );
+        connect( groupWidget, SIGNAL( toggled( bool ) ), mapper, SLOT( map() ) );
+    }
+
+    // TODO
+    emit checkReady( isReady() );
+}
+
+QStringList NetInstallPage::selectedPackages() const
+{
+    QStringList selectedPackages;
+
+    // Add all the packages for groups that are toggled in the view.
+    for ( auto it = m_groupWidgets.constBegin(); it != m_groupWidgets.constEnd(); it++ )
+    {
+        if ( it.value()->isToggled() )
+            selectedPackages += m_groups[it.key()].packages;
+    }
+
+    // Add all the packages for groups that are hidden but selected.
+    for ( const Group& group : m_groups.values() )
+    {
+        if ( group.hidden && group.selected )
+            selectedPackages += group.packages;
+    }
+
+    return selectedPackages;
+}
+
+void NetInstallPage::loadGroupList()
+{
+    QString confUrl(
+        Calamares::JobQueue::instance()->globalStorage()->value(
+            "groupsUrl" ).toString() );
+
+    KIO::Job* getJob = KIO::storedGet( confUrl, KIO::Reload, KIO::Overwrite | KIO::HideProgressInfo );
```

Comment:
As of Qt 5.6, there is the attribute enum value `QNetworkRequest::FollowRedirectsAttribute` to make your QNetworkReply follow redirects. Could this work for you?
I'm asking because KIO drags in a bucketload of dependencies that distros based on something other than Plasma might not want. If there is no reasonable way to make it work without KIO, let's use KIO, but I'd like for all of us to be sure about QNetworkAccessManager & friends not being up to the task.